e2e_test.ts

```diff
@@ -34,40 +33,47 @@ import options from "./fresh.config.ts";
  */
 const handler = await createHandler(manifest, options);
 
-function assertResponseNotFound(resp: Response) {
-  assertFalse(resp.ok);
-  assertEquals(resp.status, Status.NotFound);
+function assertHtml(resp: Response) {
+  assertInstanceOf(resp.body, ReadableStream);
+  assertEquals(
+    resp.headers.get("content-type"),
+    "text/html; charset=utf-8",
+  );
 }
 
-function assertResponseJson(resp: Response) {
-  assert(resp.ok);
-  assertNotEquals(resp.body, null);
+function assertJson(resp: Response) {
+  assertInstanceOf(resp.body, ReadableStream);
   assertEquals(resp.headers.get("content-type"), "application/json");
 }
 
-Deno.test("[e2e] GET /", async () => {
-  const resp = await handler(new Request("http://localhost"));
+function assertRedirect(response: Response, location?: string) {
+  assert(isRedirectStatus(response.status));
+  if (location !== undefined) {
+    assertStringIncludes(response.headers.get("location")!, location);
+  } else {
+    assert(response.headers.has("location"));
+  }
+}
 
-  assert(resp.ok);
+function assertXml(resp: Response) {
   assertInstanceOf(resp.body, ReadableStream);
   assertEquals(
     resp.headers.get("content-type"),
-    "text/html; charset=utf-8",
+    "application/atom+xml; charset=utf-8",
   );
-  assertEquals(resp.status, 200);
+}
+
+Deno.test("[e2e] GET /", async () => {
+  const resp = await handler(new Request("http://localhost"));
+
+  assertHtml(resp);
```

Comment:
Let's not forget to always check the response status first.

Reply:
Sorry, to be clear, the status assertion should appear before any custom assertion. The custom assertion should not include the status check unless required. E.g. A status check in `assertRedirect()` makes sense, but a status check in `assertHtml()` is beyond the scope of the assertion. I hope this makes sense.

Reply:
sorry about the misunderstanding!
extracted the status checks on all custom assertions but `assertRedirect`.